rustup-init.sh

```diff
@@ -371,49 +375,72 @@ downloader() {
     if [ "$1" = --check ]; then
         need_cmd "$_dld"
     elif [ "$_dld" = curl ]; then
-        if ! check_help_for curl --proto --tlsv1.2; then
-            echo "Warning: Not forcing TLS v1.2, this is potentially less secure"
-            curl --silent --show-error --fail --location "$1" --output "$2"
+        curl_secure_flags=$(check_curl_options)
+        if [ -n "$curl_secure_flags" ]; then
+            # shellcheck disable=SC2086
+            curl $curl_secure_flags --silent --show-error --fail --location "$1" --output "$2"
         else
-            curl --proto '=https' --tlsv1.2 --silent --show-error --fail --location "$1" --output "$2"
+            curl --silent --show-error --fail --location "$1" --output "$2"
         fi
     elif [ "$_dld" = wget ]; then
-        if ! check_help_for wget --https-only --secure-protocol; then
-            echo "Warning: Not forcing TLS v1.2, this is potentially less secure"
-            wget "$1" -O "$2"
+        wget_secure_flags=$(check_wget_options)
+        if [ -n "$wget_secure_flags" ]; then
+            # shellcheck disable=SC2086
+            wget $wget_secure_flags "$1" -O "$2"
         else
-            wget --https-only --secure-protocol=TLSv1_2 "$1" -O "$2"
+            wget "$1" -O "$2"
         fi
     else
         err "Unknown downloader"   # should not reach here
     fi
 }
 
-check_help_for() {
-    local _cmd
-    local _arg
-    local _ok
-    _cmd="$1"
-    _ok="y"
-    shift
-
-    # If we're running on OS-X, older than 10.13, then we always
-    # fail to find these options to force fallback
-    if check_cmd sw_vers; then
-        if [ "$(sw_vers -productVersion | cut -d. -f2)" -lt 13 ]; then
-            # Older than 10.13
-            echo "Warning: Detected OS X platform older than 10.13"
```

Comment:
You appear to have removed this test and not put in the equivalent anywhere else.  This test is necessary because 10.12's curl claims support for the arguments, but fails in a way which is indistinguishable from a MITM attack if they are used.

Reply:
I don't understand. I'd add new commits to test those macOS images listed
in https://docs.travis-ci.com/user/reference/osx/#macos-version.

Reply:
This is also meant to cope for users, so do not do anything travis-specific.